libvast/src/arrow_table_slice.cpp

```diff
@@ -146,10 +147,10 @@ caf::error arrow_table_slice::deserialize(caf::deserializer& source) {
     batch_ = nullptr;
     return caf::none;
   }
-  std::shared_ptr<arrow::RecordBatchReader> reader;
-  auto st = arrow::ipc::RecordBatchStreamReader::Open(&input_stream, &reader);
-  if (!st.ok())
+  auto reader_result = arrow::ipc::RecordBatchStreamReader::Open(&input_stream);
```

Comment:
Is the API following RAII principles or do we also need to invoke `RecordBatchStreamReader::Close` somewhere?

Reply:
It's just a factory function for a ref-counted RecordBatchReader. I agree that the naming is unfortunate, but we cannot change that. It's doing RAII as expected.